### PR TITLE
Adds Nesting selector

### PR DESCRIPTION
**Pull request by rachelandrew**
#### Summary

CSS Nesting is shipping in Chrome 112, and is also in Safari Beta 16.5. It introduces a new selector `&`, so I've added this under selectors.

#### Test results and supporting details

- https://chromestatus.com/feature/5800613594529792
- https://developer.apple.com/documentation/safari-release-notes/safari-16_5-release-notes#CSS

### CODE REVIEW

File: css/selectors/nesting.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "css": {
+    "selectors": {
+      "nesting": {
+        "__compat": {
+          "description": "Nesting selector (<code>&</code>)",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/nesting",
```
Comment (teoli2003) [inline css/selectors/nesting.json:7]: Not convinced by this URL (and its capitalization). 

We have [Attritube_selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors) and [Class_selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Class_selectors), …

We should follow the pattern:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/Nesting_selectors",
```

**Comment by teoli2003** [comment]
> Looks like Safari 16.5 hasn't been added as a version yet.

Shouldn't this be `preview` (not an expert here, but I think 16.5 is only in preview for the moment)

**Comment by rachelandrew** [comment]
It will need adding to browsers/safari.json as `beta`.

File: css/selectors/nesting.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            },
+            "firefox_android": "mirror",
+            "ie": {
+              "version_added": false
+            },
+            "oculus": "mirror",
+            "opera": "mirror",
+            "opera_android": "mirror",
+            "safari": {
+              "version_added": "16.5"
```
Comment (queengooborg) [inline css/selectors/nesting.json:26]: This should do the trick!

```suggestion
              "version_added": "preview"
```

**Review by queengooborg — APPROVED** [review]

**Pull request merged**